mobile_verifier/tests/test_poc_shares.rs

```diff
@@ -0,0 +1,158 @@
+use std::{ops::Range, pin::pin};
+
+use chrono::{DateTime, Duration, Utc};
+use file_store::{
+    heartbeat::{CellHeartbeat, CellHeartbeatIngestReport},
+    speedtest::CellSpeedtest,
+};
+use futures::stream::{self, StreamExt};
+use helium_crypto::PublicKeyBinary;
+use mobile_verifier::{
+    heartbeats::{Heartbeat, HeartbeatReward},
+    reward_shares::PocShares,
+    speedtests::{SpeedtestAverages, SpeedtestRollingAverage},
+    HasOwner,
+};
+use rust_decimal_macros::dec;
+use sqlx::PgPool;
+
+#[derive(Copy, Clone)]
+struct AllOwnersValid;
+
+#[async_trait::async_trait]
+impl HasOwner for AllOwnersValid {
+    type Error = anyhow::Error;
+
+    async fn has_owner(&self, _address: &PublicKeyBinary) -> Result<bool, Self::Error> {
+        Ok(true)
+    }
+}
+
+fn heartbeats<'a>(
+    num: usize,
+    start: DateTime<Utc>,
+    hotspot_key: &'a PublicKeyBinary,
+    cbsd_id: &'a str,
+    lon: f64,
+    lat: f64,
+) -> impl Iterator<Item = CellHeartbeatIngestReport> + 'a {
+    (0..num).map(move |i| {
+        let report = CellHeartbeat {
+            pubkey: hotspot_key.clone(),
+            lon,
+            lat,
+            operation_mode: true,
+            cbsd_id: cbsd_id.to_string(),
+            // Unused:
+            hotspot_type: String::new(),
+            cell_id: 0,
+            timestamp: DateTime::<Utc>::MIN_UTC,
+            cbsd_category: String::new(),
+        };
+        CellHeartbeatIngestReport {
+            report,
+            received_timestamp: start + Duration::hours(i as i64),
+        }
+    })
+}
+
+fn bytes_per_s(mbps: u64) -> u64 {
+    mbps * 125000
+}
+
+fn acceptable_speedtests(
+    num: usize,
+    owner: &'_ PublicKeyBinary,
+    timestamp_start: DateTime<Utc>,
+) -> impl Iterator<Item = CellSpeedtest> + '_ {
+    (0..num).map(move |i| {
+        acceptable_speedtest(owner, timestamp_start + Duration::hours((i * 12) as i64))
+    })
+}
+
+fn acceptable_speedtest(owner: &PublicKeyBinary, timestamp: DateTime<Utc>) -> CellSpeedtest {
+    CellSpeedtest {
+        pubkey: owner.clone(),
+        timestamp,
+        upload_speed: bytes_per_s(10),
+        download_speed: bytes_per_s(100),
+        latency: 25,
+        // Unused:
+        serial: String::new(),
+    }
+}
+
+async fn rewards(
```

Comment:
Whilst this function does some of what the actual rewarder does and tests key functions of that flow, an integration test should be using the actual rewarder flow itself rather than shadowing it.  Its fine to have prep functions to generate the data etc but i think this rewards function negates a lot of the point of these being integration tests